src/Capability/Derive.hs

```diff
@@ -9,21 +9,41 @@
 {-# LANGUAGE TypeApplications #-}
 {-# LANGUAGE TypeOperators #-}
 
-module Capability.Context where
+module Capability.Derive where
 
 import Capability.Constraints
 import Data.Coerce (Coercible)
 import Unsafe.Coerce (unsafeCoerce)
 
--- | Execute the given action with an additional @inner@ capability derived from
--- current context via the @t@ wrapper, retaining the list @cs@ of capabilities.
-context ::
+-- | @'derive' \@t \@derived \@ambient act@ is used to run @act@ when the
+-- capabilities required by @act@ are not necessarily the same as those
+-- available.
+--
+-- @'derive' \@t \@derived \@ambient act@ runs @act@ by providing both the
+-- capabilities in @derived@ and @ambient@. The difference is that @ambient@
+-- capabilities are assumed to be available, whereas @derived@ instances are
+-- provided by @t@.
```

Comment:
There's a bit of duplication between those sentences. Maybe we could just drop the first sentence?

Reply:
Not sure. It's really on purpose. The first paragraph answers the question of: “why”, and the second paragraph answers “how”.

Is it grating to read? I'm really, really ok with duplication. Unless it really sounds horrible :slightly_smiling_face: .

Reply:
I see, okay with that spelled out I can see how those two sentences serve different purposes. I personally find it looks a bit scary when rendered. Together with the signature we're looking at three lines of complicated looking code without an introduction in plain English.

What do you think of this as a first sentence instead?
```
-- | Runs an action that requires additional capabilities.
```

Reply:
I've changed the first sentence.